jina/drivers/encode.py

```diff
@@ -1,25 +1,84 @@
 __copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
 __license__ = "Apache-2.0"
 
-from . import BaseExecutableDriver
+from typing import Optional
 
-if False:
-    from ..types.sets import DocumentSet
+from . import BaseExecutableDriver
+from ..types.sets import DocumentSet
 
 
 class BaseEncodeDriver(BaseExecutableDriver):
-    """Drivers inherited from this Driver will bind :meth:`craft` by default """
+    """Drivers inherited from this Driver will bind :meth:`encode` by default """
 
-    def __init__(self, executor: str = None, method: str = 'encode', *args, **kwargs):
+    def __init__(self,
+                 executor: str = None,
+                 method: str = 'encode',
+                 *args, **kwargs):
         super().__init__(executor, method, *args, **kwargs)
 
 
 class EncodeDriver(BaseEncodeDriver):
-    """Extract the chunk-level content from documents and call executor and do encoding
+    """Extract the content from documents and call executor and do encoding
     """
 
-    def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
-        contents, docs_pts, bad_docs = docs.all_contents
+    class CacheDocumentSet:
+        """Helper class to accumulate documents from differents document Set in a single DocumentSet
+         to help guarantee that the encoder driver can consume documents in fixed batch sizes to allow
+         the EncoderExecutors to leverage its batching abilities.
+
+         It is useful to have batching even when chunks are involved"""
```

Comment:
i have a feeling this should be in types? or even the `__add__` of `DocumentSet`? feel that this should be more native and general. Will look at this later

Reply:
I also had the feeling, but I was being conservative as per how universal this may be needed, and that's why I kept it inside `EncodeDriver`